E2E/tests/reenterGame.test.ts

```diff
@@ -0,0 +1,113 @@
+Feature('Reenter game')
+
+const { I, login, game } = inject();
+import { ChessPiece } from "../test_data/ChessPieces";
+import username = require("../test_data/usernames");
+import ChessTile = require("../test_data/chessTiles");
+import { ChessPieceValue } from "../test_data/chessPieces";
+
+Scenario('Player, who was skipped from game by others, leaves and returns back to game room', async () => {
+  login.firstVoterLogin(username.user1);
+  let url = await I.grabCurrentUrl();
+  session(username.user2, () => {
+    login.voterLoginIntoCreatedGameRoom(url, username.user2);
+  });
+  I.click(game.locator.buttons.skipOtherPlayer(username.user2));
+  session(username.user2, () => {
+    game.navigateBackAndForward();
+    game.gameInProgressStatus();
+    I.seeElement(game.locator.buttons.skipButtonHighlighted);
+  });
+  game.vote(ChessPiece.knight, ChessTile.e4);
+  I.waitForText('Game complete');
+});
+
+Scenario('Player, who was skipped from voting by other player, refreshes browser while game is still in progress', async () => {
+  login.firstVoterLogin(username.user1);
+  let url = await I.grabCurrentUrl();
+  session(username.user2, () => {
+    login.voterLoginIntoCreatedGameRoom(url, username.user2);
+  });
+  I.click(game.locator.buttons.skipOtherPlayer(username.user2));
+  session(username.user2, () => {
+    I.refreshPage();
+    game.gameInProgressStatus();
+    I.seeElement(game.locator.buttons.skipButtonHighlighted);
+  });
+});
+
+Scenario('Player who already skipped the move refreshes browser when game is completed', async () => {
+  login.firstVoterLogin(username.user1);
+  let url = await I.grabCurrentUrl();
+  session(username.user2, () => {
+    login.voterLoginIntoCreatedGameRoom(url, username.user2);
+  });
+  game.vote(ChessPiece.rook, ChessTile.d3);
+  session(username.user2, () => {
+    game.skipMove();
+    I.refreshPage();
+    I.seeElement(game.locator.buttons.skipButtonHighlighted);
+    I.waitForText('Game complete');
+  });
+});
+  
+Scenario('Player leaves and returns back to game room, after game is completed', async () => {
+  login.firstVoterLogin(username.user1);
+  let url = await I.grabCurrentUrl();
+  session(username.user2, () => {
+    login.voterLoginIntoCreatedGameRoom(url, username.user2);
+    game.voteAndCheckThatVoteIsVisible(ChessPiece.pawn, ChessTile.a1, ChessPieceValue.pawn);
+  });
+  game.voteAndCheckThatVoteIsVisible(ChessPiece.knight, ChessTile.e4, ChessPieceValue.knight);
+  I.waitForText('Game complete');
+  const actualFinalScoreBeforeNavigating = Number((await I.grabTextFrom(game.locator.playersList.totalSP)).slice(16, -3));
+  game.navigateBackAndForward();
+  I.waitForText('Game complete');
+  game.voteIsVisible(ChessPiece.pawn, ChessTile.a1, ChessPieceValue.pawn);
+  game.voteIsVisible(ChessPiece.knight, ChessTile.e4, ChessPieceValue.knight);
+ 
+  const actualFinalScoreAfterNavigating = Number((await I.grabTextFrom(game.locator.playersList.totalSP)).slice(16, -3));
```

Comment:
Move SP parsing from string into a separate file / function.

Also, consider using regexp for this kind of parsing, it would make it much more readable
```
const str = "Game complete - 5 SP"
const regexp = /Game complete - (?<spValue>\d+)+ SP/;
const { groups } = str.match(regexp); //groups.spValue = 5
```